### PR TITLE
Add intent for running from a script file

### PR DESCRIPTION
**Pull request by johnmccutchan**
Needed for hot reloading support

@jason-simmons

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Do we still need to do anything to gate intent handling to be only Debug only?  Or is that no longer a concern with AOT?  (fyi @abarth)

File: sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java [hunk 1]
```diff
@@ -135,13 +135,19 @@ protected void onNewIntent(Intent intent) {
...
+            String packageRoot = intent.getStringExtra("packages");
+            if ((file != null) && (packageRoot != null)) {
+                mView.runFromFile(file, packageRoot);
+            } else {
+                mView.runFromBundle(intent.getDataString(),
+                                    intent.getStringExtra("snapshot"));
+                String route = intent.getStringExtra("route");
+
+                if (route != null)
+                    mView.pushRoute(route);
```
Comment (abarth) [inline sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java]: Shouldn't this logic be shared between the file and bundle code paths?

**Comment by abarth** [comment]
I don't understand how running from a file works.  If you run from a file, you won't have an asset bundle, which means you won't have basic things like fonts and icons.

File: sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java (hunk 1 above)
Comment (johnmccutchan) [inline sky/shell/platform/android/org/domokit/sky/shell/SkyActivity.java]: Done.

**Comment by johnmccutchan** [comment]
@abarth Yes, currently we are running without one specified. Bundles are next on my TODO list.

**Comment by abarth** [comment]
We talked it over.  This intend handling is just a stepping stone towards the final design and will probably be removed at some point.

LGTM

Please add a comment to the intent handling describing in a few sentences what design we're headed towards so folks don't start building around run-from-files if it's going to be removed eventually.

**Comment by johnmccutchan** [comment]
I've added a comment about the plan around loading from a file.

**Pull request merged**